check/src/kindcheck.rs

```diff
@@ -192,41 +192,46 @@ impl<'a> KindCheck<'a> {
                 Ok((kind, Type::app(ctor, new_args)))
             }
             Type::Variant(ref row) => {
-                let row = try!(row.row_iter()
+                let row = row.row_iter()
                     .map(|field| {
-                        let (kind, typ) = try!(self.kindcheck(&field.typ));
+                        let (kind, typ) = self.kindcheck(&field.typ)?;
                         let type_kind = self.type_kind();
-                        try!(self.unify(&type_kind, kind));
+                        self.unify(&type_kind, kind)?;
                         Ok(Field {
                             name: field.name.clone(),
                             typ: typ,
                         })
                     })
-                    .collect());
-                Ok((self.type_kind(), Type::variant(row)))
+                    .collect();
+
+                #[cfg_attr(rustfmt, rustfmt_skip)]
+                Ok((self.type_kind(), Type::variant(try!(row))))
```

Comment:
So rustfmt automatically rewrites `try!` now? Maybe it is better to add type annotations to help out `.collect::<Result<_, _>>()`?